gbpservice/nfp/configurator/agents/loadbalancer_v2.py

```diff
@@ -0,0 +1,978 @@
+
+        """
+        arg_dict = {'context': context,
+                    'old_health_monitor': old_healthmonitor,
+                    'health_monitor': healthmonitor
+                    }
+        self._send_event(lb_constants.EVENT_UPDATE_HEALTH_MONITOR,
+                         arg_dict, serialize=True,
+                         binding_key=healthmonitor['pool']['id'],
+                         key=healthmonitor['id'])
+
+    def delete_healthmonitor(self, context, healthmonitor):
+        """Enqueues event for worker to process delete health monitor request.
+
+        :param context: RPC context
+        :param health_monitor: health_monitor resource to be deleted
+        :param pool_id: pool_id to which health monitor is associated
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'health_monitor': healthmonitor
+                    }
+        self._send_event(lb_constants.EVENT_DELETE_HEALTH_MONITOR,
+                         arg_dict, serialize=True,
+                         binding_key=healthmonitor['pool']['id'],
+                         key=healthmonitor['id'])
+
+    # TODO(jiahao): copy from v1 agent need to review
+    def agent_updated(self, context, payload):
+        """Enqueues event for worker to process agent updated request.
+
+        :param context: RPC context
+        :param payload: payload
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'payload': payload}
+        self._send_event(lb_constants.EVENT_AGENT_UPDATED, arg_dict)
+
+
+"""Implements event handlers and their helper methods.
+
+Object of this class is registered with the event class of core service
+controller. Based on the event key, handle_event method of this class is
+invoked by core service controller.
+
+"""
+
+
+class LBaaSEventHandler(agent_base.AgentBaseEventHandler,
+                        nfp_poll.PollEventDesc):
+    instance_mapping = {}
+
+    def __init__(self, sc, drivers, rpcmgr):
+        self.sc = sc
+        self.drivers = drivers
+        self.rpcmgr = rpcmgr
+        self.plugin_rpc = LBaasRpcSender(sc)
+
+    def _get_driver(self, driver_name):
+        """Retrieves service driver object based on service type input.
+
+        Currently, service drivers are identified with service type. Support
+        for single driver per service type is provided. When multi-vendor
+        support is going to be provided, the driver should be selected based
+        on both service type and vendor name.
+
+        :param service_type: Service type - loadbalancer
+
+        Returns: Service driver instance
+
+        """
+        driver = lb_constants.SERVICE_TYPE + driver_name
+        return self.drivers[driver]
+
+    def _root_loadbalancer_id(self, obj_type, obj_dict):
+        """Returns the loadbalancer id this instance is attached to."""
+
+        try:
+            # For Mitaka
+            if obj_type == 'loadbalancer':
```

Comment:
@pkharat To be more specific, do you mean add something like:
In lb_constants, add:
LOADBALANCER = 'loadbalancer'
LISTENER = 'listener'
...etc
And change this line 495 to:
if obj_type == lb_constants.LOADBALANCER

Am I correct?